extensions/wikia/Staging/js/Staging.js

```diff
@@ -1,24 +1,26 @@
-if ( window.wgStagingEnvironment ) {
-	(function( document ){
+if (window.wgStagingEnvironment) {
+	(function (document) {
 		'use strict';
-
-		var stagingEnvName = window.location.hostname.split( '.' )[0],
-			stagingURLPart = '://' + stagingEnvName + '.',
-			links = document.getElementsByTagName('a' ),
-			i = 0,
-			//match all occurrences of wikia.com between :// and /
-			//use case: links in share icons have wikia.com but as URL parameter not domain name
-			pattern=/^\w*:?\/\/[^\/]*wikia\.com/gi,
-			href;
-
-		for ( ; i < links.length; i++ ) {
-			href = links[i].getAttribute('href');
-
-			pattern.lastIndex = 0;
-			if ( href && !links[i].classList.contains('final-url') && href.indexOf( stagingURLPart ) === -1 && pattern.test(href)) {
-				links[i].setAttribute( 'href', href.replace( '://', stagingURLPart ) );
+		var stagingEnvName = window.location.hostname.match(/([^\.]+)\.wikia\.com/);
+		if (stagingEnvName) {
+			stagingEnvName = stagingEnvName[1];
+			var links = document.getElementsByTagName('a'),
+				i = 0,
+				href,
+				hostname;
+			for (; i < links.length; i++) {
+				href = links[i].href;
+				hostname = links[i].hostname;
+				if (
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

[project-id]: wikia:app